front/src/applications/operationalStudies/components/MacroEditor/OsrdNgeSync.tsx

```diff
@@ -0,0 +1,918 @@
+import { isNil, sortBy, uniqBy } from 'lodash';
+import { Layout } from 'webcola';
+import type { Node as ColaNode } from 'webcola';
+
+import type {
+  MacroNodeResponse,
+  ScenarioResponse,
+  SearchPayload,
+  SearchQuery,
+  SearchResultItemOperationalPoint,
+  TrainScheduleResult,
+} from 'common/api/osrdEditoastApi';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+import handleOperation, {
+  DEFAULT_TRAINRUN_FREQUENCIES,
+  DEFAULT_TRAINRUN_FREQUENCY,
+} from './ngeToOsrd';
+import { addDurationToDate } from './utils';
+import {
+  type PortDto,
+  type TimeLockDto,
+  type TrainrunSectionDto,
+  type TrainrunCategory,
+  type TrainrunTimeCategory,
+  type NetzgrafikDto,
+  type LabelGroupDto,
+  PortAlignment,
+  type NGEEvent,
+  type NodeDto,
+} from '../NGE/types';
+
+const TRAINRUN_CATEGORY_HALTEZEITEN = {
+  HaltezeitIPV: { haltezeit: 0, no_halt: false },
+  HaltezeitA: { haltezeit: 0, no_halt: false },
+  HaltezeitB: { haltezeit: 0, no_halt: false },
+  HaltezeitC: { haltezeit: 0, no_halt: false },
+  HaltezeitD: { haltezeit: 0, no_halt: false },
+  HaltezeitUncategorized: { haltezeit: 0, no_halt: false },
+};
+
+const DEFAULT_LABEL_GROUP: LabelGroupDto = {
+  id: 1,
+  name: 'Default',
+  labelRef: 'Trainrun',
+};
+
+const DEFAULT_TRAINRUN_CATEGORY: TrainrunCategory = {
+  id: 1, // In NGE, Trainrun.DEFAULT_TRAINRUN_CATEGORY
+  order: 0,
+  name: 'Default',
+  shortName: '', // TODO: find a better way to hide this in the graph
+  fachCategory: 'HaltezeitUncategorized',
+  colorRef: 'EC',
+  minimalTurnaroundTime: 0,
+  nodeHeadwayStop: 0,
+  nodeHeadwayNonStop: 0,
+  sectionHeadway: 0,
+};
+
+const DEFAULT_TRAINRUN_TIME_CATEGORY: TrainrunTimeCategory = {
+  id: 0, // In NGE, Trainrun.DEFAULT_TRAINRUN_TIME_CATEGORY
+  order: 0,
+  name: 'Default',
+  shortName: '7/24',
+  dayTimeInterval: [],
+  weekday: [1, 2, 3, 4, 5, 6, 7],
+  linePatternRef: '7/24',
+};
+
+const DEFAULT_DTO: NetzgrafikDto = {
+  resources: [],
+  nodes: [],
+  trainruns: [],
+  trainrunSections: [],
+  metadata: {
+    netzgrafikColors: [],
+    trainrunCategories: [DEFAULT_TRAINRUN_CATEGORY],
+    trainrunFrequencies: [...DEFAULT_TRAINRUN_FREQUENCIES],
+    trainrunTimeCategories: [DEFAULT_TRAINRUN_TIME_CATEGORY],
+  },
+  freeFloatingTexts: [],
+  labels: [],
+  labelGroups: [],
+  filterData: {
+    filterSettings: [],
+  },
+};
+
+const DEFAULT_TIME_LOCK: TimeLockDto = {
+  time: null,
+  consecutiveTime: null,
+  lock: false,
+  warning: null,
+  timeFormatter: null,
+};
+
+type NodeIndex = { node: MacroNodeResponse; saved: boolean };
```

Comment:
nit: why not flatten the type ?
```suggestion
type NodeIndex = MacroNodeResponse & { saved: boolean };
```

Reply:
I considered the  `saved` attribut  as a metadata, that's why I created a wrapper on it.

It's possible to flatten this type, but then I need to use the lodash `omit` function in some part of the code , like for example when calling `apiUpdateNode` or `apiCreateNode`.

Both solution are good for me. Do you want me to change it ?